src/lib/queries/lexical-query.js

```diff
@@ -63,15 +58,12 @@ export default class LexicalQuery extends Query {
         if (this.annotatedHomonym) {
           this.homonym = Homonym.disambiguate(this.homonym, [this.annotatedHomonym])
         }
-        this.ui.addMessage(this.ui.l10n.messages.TEXT_NOTICE_MORPHDATA_READY)
+        LexicalQuery.evt.MORPH_DATA_READY.pub()
       } else {
         if (this.annotatedHomonym) {
           this.homonym = this.annotatedHomonym
         } else {
-          this.ui.addImportantMessage(this.ui.l10n.messages.TEXT_NOTICE_MORPHDATA_NOTFOUND)
-          // Need to notify a UI controller that there is no morph data on this word in an analyzer
-          // However, controller may not have `morphologyDataNotFound()` implemented, so need to check first
-          if (this.ui.morphologyDataNotFound) { this.ui.morphologyDataNotFound(true) }
+          LexicalQuery.evt.MORPH_DATA_NOT_FOUND.pub()
```

Comment:
I am not sure that I correctly understood the meaning of this message.
But this condition doesn't mean that the morph data didn't found.
It could  mean that:
  a) AlpheiosTreebank adapter is not available
  b) Alpheios treebank is not configured 
  c) the word doesn't have additional tag word-ref and word couldn't be search by ref
   
In this case we get all avalable morph data about the not according to the word-ref.

Reply:
I just used an existing name of the message (`TEXT_NOTICE_MORPHDATA_NOTFOUND`) and of the function (`morphologyDataNotFound`) that existed in the previous code. But I haven't touched this code for a while so have no clear understanding of what exactly is going on here.

Can you suggest a better name for it? Does it mean at this stage that we have no morph data available? Then maybe we could use MORPH_DATA_NOT_AVAILABLE or something like that? What do you think?

Reply:
This (retrieval of morph data from an aligned treebank) is a pretty specific case and is really still prototype level code that might change. I think making the event name more specific is probably a good idea.  I would suggest perhaps TREEBANK_DATA_READY and TREEBANK_DATA_NOTAVAILABLE. (I think only the pwa controller currently has the morphDataNotFound method implemented )